tests/test_BlkDiagMatrix.py

```diff
@@ -0,0 +1,910 @@
+                      [ 0.74976559,  0.56202548]]),
+            np.array([[-0.18610643,  0.74976559],
+                      [ 0.74976559,  0.56202548]]),
+            np.array([[-0.31969785]]),
+            np.array([[-0.31969785]]),
+            np.array([[-0.41647978]]),
+            np.array([[-0.41647978]]),
+            np.array([[-0.47055057]]),
+            np.array([[-0.47055057]])
+        ]
+
+        blk_c = self.blk_a + self.blk_b
+        self.allallfunc(result, blk_c)
+
+        blk_c = self.blk_a.add(self.blk_b)
+        self.allallfunc(result, blk_c)
+
+    def testBlkDiagMatrixSub(self):
+        result = [
+            np.array([[-0.03199698, -0.00517155,  0.08213249,  0.00745266],
+                      [-0.00517155,  0.04924602, -0.02805815, -0.01263298],
+                      [ 0.08213249, -0.02805815, -0.04744928,  0.07403869],
+                      [ 0.00745266, -0.01263298,  0.07403869, -0.18387143]]),
+            np.array([[-0.0282204 , -0.06325738,  0.06453211],
+                      [-0.06325738,  0.02472711,  0.03180649],
+                      [ 0.06453211,  0.03180649, -0.08552209]]),
+            np.array([[-0.0282204 , -0.06325738,  0.06453211],
+                      [-0.06325738,  0.02472711,  0.03180649],
+                      [ 0.06453211,  0.03180649, -0.08552209]]),
+            np.array([[-0.00139116, -0.0858976 ,  0.00933522],
+                      [-0.0858976 , -0.03649255,  0.06719452],
+                      [ 0.00933522,  0.06719452, -0.15479544]]),
+            np.array([[-0.00139116, -0.0858976 ,  0.00933522],
+                      [-0.0858976 , -0.03649255,  0.06719452],
+                      [ 0.00933522,  0.06719452, -0.15479544]]),
+            np.array([[ 0.02524017, -0.07616898],
+                      [-0.07616898, -0.09700953]]),
+            np.array([[ 0.02524017, -0.07616898],
+                      [-0.07616898, -0.09700953]]),
+            np.array([[ 0.04338129, -0.0453599 ],
+                      [-0.0453599 , -0.15307969]]),
+            np.array([[ 0.04338129, -0.0453599 ],
+                      [-0.0453599 , -0.15307969]]),
+            np.array([[ 0.05112375, -0.00072091],
+                      [-0.00072091, -0.20420688]]),
+            np.array([[ 0.05112375, -0.00072091],
+                      [-0.00072091, -0.20420688]]),
+            np.array([[0.04867057]]),
+            np.array([[0.04867057]]),
+            np.array([[0.03697756]]),
+            np.array([[0.03697756]]),
+            np.array([[0.01732433]]),
+            np.array([[0.01732433]])
+        ]
+
+        blk_c = self.blk_a - self.blk_b
+        self.allallfunc(result, blk_c)
+
+        blk_c = self.blk_a.sub(self.blk_b)
+        self.allallfunc(result, blk_c)
+
+    def testBlkDiagMatrixApply(self):
+
+        mean_coeff = np.array([
+            [ 4.53531036e-04],
+            [ 2.29341625e-04],
+            [ 9.65040155e-05],
+            [-1.19442447e-04],
+            [ 0.00000000e+00],
```

Comment:
Maybe not something with so many zeros?

Reply:
Sure, I just used the original tests.

Reply:
Right, I understand. Following our discussion, maybe we postpone overhauling these tests (possibly making them less hard-coded).

Reply:
Sure, here is how far I got with that idea yesterday:

https://github.com/garrettwrong/ASPIRE-Python/commit/103c6cba33c6379051f98201f4cec403426103aa

Reply:
#156 should cover when the time is right.

Reply:
>  garrettwrong@103c6cb

Haven't had time to look through in detail, but it looks good.